cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/config/reader/ContextConfigurationReader.java

```diff
@@ -59,19 +58,21 @@
     public ContextConfigurationReader(Path configRoot) throws InvalidConfigurationException {
 
         contextFile = configRoot.resolve(ConfigurationConstants.CONTEXT_CONFIG_FILENAME);
-        filePath = contextFile.toAbsolutePath().toString();
-
-        try {
-            readConfiguration();
-        } catch (InvalidConfigurationException e) {
-            configRoot = configRoot.resolve("src/main/templates");
+        if (!Files.exists(contextFile)) {
+            configRoot = configRoot.resolve(ConfigurationConstants.TEMPLATE_RESOURCE_FOLDER);
             contextFile = configRoot.resolve(ConfigurationConstants.CONTEXT_CONFIG_FILENAME);
-            filePath = contextFile.toAbsolutePath().toString();
-            readConfiguration();
+            if (!Files.exists(contextFile)) {
+                throw new InvalidConfigurationException(contextFile, "Could not read context configuration file.");
```

Comment:
better: Could not find configuration file as read would imply that you have found it, but you are unable to read it.